sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -619,6 +623,122 @@ public void teardown() throws Exception {
     }
   }
 
+  /** A {@link PTransform} to write to a JDBC datasource. */
+  @AutoValue
+  public abstract static class WriteIterable<T>
+          extends PTransform<PCollection<Iterable<T>>, PDone> {
+    @Nullable abstract DataSourceConfiguration getDataSourceConfiguration();
+    @Nullable abstract String getStatement();
+    @Nullable abstract PreparedStatementSetter<T> getPreparedStatementSetter();
+
+    abstract Builder<T> toBuilder();
+
+    @AutoValue.Builder
+    abstract static class Builder<T> {
+      abstract Builder<T> setDataSourceConfiguration(DataSourceConfiguration config);
+      abstract Builder<T> setStatement(String statement);
+      abstract Builder<T> setPreparedStatementSetter(PreparedStatementSetter<T> setter);
+
+      abstract WriteIterable<T> build();
+    }
+
+    public WriteIterable<T> withDataSourceConfiguration(DataSourceConfiguration config) {
+      return toBuilder().setDataSourceConfiguration(config).build();
+    }
+    public WriteIterable<T> withStatement(String statement) {
+      return toBuilder().setStatement(statement).build();
+    }
+    public WriteIterable<T> withPreparedStatementSetter(PreparedStatementSetter<T> setter) {
+      return toBuilder().setPreparedStatementSetter(setter).build();
+    }
+
+    @Override
+    public PDone expand(PCollection<Iterable<T>> input) {
+      checkArgument(
+          getDataSourceConfiguration() != null, "withDataSourceConfiguration() is required");
+      checkArgument(getStatement() != null, "withStatement() is required");
+      checkArgument(
+          getPreparedStatementSetter() != null, "withPreparedStatementSetter() is required");
+
+      input.apply(ParDo.of(new WriteIterableFn<T>(this)));
+      return PDone.in(input.getPipeline());
+    }
+
+    private static class WriteIterableFn<T> extends DoFn<Iterable<T>, Void> {
+      private static final int DEFAULT_BATCH_SIZE = 1000;
```

Comment:
In the case of `WriteIteratble` I would not use batch here.

Reply:
Just to be clear: this batch size is when we have a single element. Here with a `PCollection<Iterable>`, the batch size is basically the size of the `Iterable` element.